src/main/java/com/flightstats/hub/dao/ReadOnlyDocumentationDao.java

```diff
@@ -0,0 +1,24 @@
+package com.flightstats.hub.dao;
+
+public class ReadOnlyDocumentationDao implements DocumentationDao {
+    private final DocumentationDao delegate;
+
+    public ReadOnlyDocumentationDao(DocumentationDao delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public String get(String channel) {
+        return delegate.get(channel);
+    }
+
+    @Override
+    public boolean upsert(String channel, byte[] content) {
```

Comment:
I'd rather throw them from the concrete class that knows enough to explain _why_ it's violating it's Is-A contract.  This is really a safety valve.  The best solution is probably to re-factor hub along read/write seams so that writes only ever go to explicit writers and the exception case couldn't occur, but this was a pretty invasive refactor when I started going that route.

Reply:
Fair enough :) This is an improvement on what's there. There are a LOT of times I'll do something that I know is at least a step in the right direction even if I know there are weakness. Then I'll come back to it with a future PR when I've had time to ponder how to approach it. Don't let perfect be the enemy of good and all that :)